src/utils/test-util.ts

```diff
@@ -149,6 +149,46 @@ export async function getIssuedCredential(
   assert(credentialList.length <= 1);
   return credentialList[0];
 }
+/**
+ * Get issued credential with specific role
+ * This extends the default credential retrieval to filter by role
+ */
+export async function getIssuedCredentialByRole(
+  client: SignifyClient.SignifyClient,
+  issuerAID: any,
+  recipientAID: any,
+  schema: string,
+  role: string
+) {
+  // Get all credentials matching the basic criteria
+  const credentials = await client.credentials().list({
+    filter: {
+      '-i': issuerAID.prefix,
+      '-s': schema,
+      '-a-i': recipientAID.prefix,
+    },
+  });
+
+  // If no credentials found, return null
+  if (!credentials || credentials.length === 0) {
+    return null;
+  }
+
+  console.log(`Looking for role: "${role}"`);
+  console.log(`Found ${credentials.length} credentials with roles:`);
+  credentials.forEach((cred, index) => {
+    console.log(
+      `  ${index + 1}. Role: "${cred.sad.a.engagementContextRole}" | SAID: ${cred.sad.d}`
+    );
+  });
+
+  // Find the credential with the matching role
+  const matchingCred = credentials.find(
+    (cred) => cred.sad.a.engagementContextRole === role
```

Comment:
good first shot. did you try to use the filter object above to include something like -a-engagementContextRole? if that would work then this function could be refactored to be more generic by just taking a filter object as a parameter allowing callers to pass any set of filters they want